tests/test_odp_manager.py

```diff
@@ -0,0 +1,299 @@
+        manager.fetch_qualified_segments('user1', [])
+        mock_logger.debug.assert_called_once_with(Errors.ODP_NOT_ENABLED)
+
+        manager.update_odp_config('valid', 'host', [])
+        self.assertIsNone(manager.odp_config.get_api_key())
+        self.assertIsNone(manager.odp_config.get_api_host())
+
+        # these call should be dropped gracefully with None
+        manager.identify_user('user1')
+
+        self.assertRaisesRegex(optimizely_exception.OdpNotEnabled, Errors.ODP_NOT_ENABLED,
+                               manager.send_event, 't1', 'a1', {}, {})
+
+        self.assertIsNone(manager.event_manager)
+        self.assertIsNone(manager.segment_manager)
+
+    def test_fetch_qualified_segments(self):
+        mock_logger = mock.MagicMock()
+        segment_manager = OdpSegmentManager(OdpConfig(), LRUCache(1000, 1000),
+                                            ZaiusGraphQLApiManager(), mock_logger)
+
+        manager = OdpManager(False, 10, 20, segment_manager, None, mock_logger)
+
+        with mock.patch.object(segment_manager, 'fetch_qualified_segments') as mock_fetch_qualif_segments:
+            manager.fetch_qualified_segments('user1', [])
+
+        mock_logger.debug.assert_not_called()
+        mock_fetch_qualif_segments.assert_called_once_with('fs_user_id', 'user1', [])
+
+    def test_identify_user_datafile_not_ready(self):
+        mock_logger = mock.MagicMock()
+        event_manager = OdpEventManager(OdpConfig(), mock_logger)
+        event_manager.start()
+
+        manager = OdpManager(False, 10, 20, None, event_manager, mock_logger)
+
+        with mock.patch.object(event_manager, 'identify_user') as mock_identify_user:
+            manager.identify_user('user1')
+
+        mock_identify_user.assert_called_once_with('user1')
+
+    def test_identify_user_odp_integrated(self):
+        mock_logger = mock.MagicMock()
+        event_manager = OdpEventManager(OdpConfig(), mock_logger, ZaiusRestApiManager())
+        event_manager.start()
+
+        manager = OdpManager(False, 10, 20, None, event_manager, mock_logger)
+        manager.update_odp_config('key1', 'host1', [])
+
+        with mock.patch.object(event_manager, 'dispatch') as mock_dispatch_event:
+            manager.identify_user('user1')
+
+        mock_dispatch_event.assert_called_once()
+        self.assertEqual(mock_dispatch_event.call_args[0][0].type, 'fullstack')
+        self.assertEqual(mock_dispatch_event.call_args[0][0].action, 'identified')
+        self.assertEqual(mock_dispatch_event.call_args[0][0].identifiers, {'fs_user_id': 'user1'})
+        self.assertGreater(len(mock_dispatch_event.call_args[0][0].data['idempotence_id']), 0)
+        self.assertEqual(mock_dispatch_event.call_args[0][0].data['data_source_type'], 'sdk')
+        self.assertEqual(mock_dispatch_event.call_args[0][0].data['data_source'], 'python-sdk')
+        self.assertEqual(mock_dispatch_event.call_args[0][0].data['data_source_version'], '4.1.0')
```

Comment:
Clean up using `assert_called_once_with`, `mock.ANY` and `version.__version__`

Reply:
done